content/concepts/did-ddo.md

```diff
@@ -1,173 +1,641 @@
+| **`allowRawAlgorithm`**                    | `boolean`                             | **✓**    | If `true`, any passed raw text will be allowed to run. Useful for an algorithm drag & drop use case, but increases risk of data escape through malicious user input. Should be `false` by default in all implementations.                                                                                                            |
+| **`allowNetworkAccess`**                   | `boolean`                             | **✓**    | If `true`, the algorithm job will have network access.                                                                                                                                                                                                                                                                               |
+| **`publisherTrustedAlgorithmPublishers `** | Array of `string`                     | **✓**    | If empty, then any published algorithm is allowed. Otherwise, only published algorithms by some publishers are allowed.                                                                                                                                                                                                              |
+| **`publisherTrustedAlgorithms `**          | Array of `publisherTrustedAlgorithms` | **✓**    | If empty, then any published algorithm is allowed. (see below).                                                                                                                                                                                                                                                                      |
 
-Only credentials that can be proven are supported. This includes Ethereum public addresses, and (in the future) W3C Verifiable Credentials and more.
+The `publisherTrustedAlgorithms ` is an array of objects with the following structure:
 
-Ocean also supports `"deny"` credentials: if a consumer has any of these credentials, they cannot access the resource.
+| Attribute                      | Type     | Required | Description                                                               |
+| ------------------------------ | -------- | -------- | ------------------------------------------------------------------------- |
+| **`did`**                      | `string` | **✓**    | The DID of the algorithm which is trusted by the publisher.               |
+| **`filesChecksum`**            | `string` | **✓**    | Hash of algorithm's `files` section (as `string`).                        |
+| **`containerSectionChecksum`** | `string` | **✓**    | Hash of algorithm's `metadata.algorithm.container` section (as `string`). |
```

Comment:
I think it would be good to differentiate `string` from `HexStr`. The way things are currently worded, it's unclear whether the `filesChecksum` and `containerSectionChecksum` are encoded as `HexStr` (with a `0x` prefix) or not. And for the `files` attribute, it only becomes clear that it should be a `HexStr` after reading the example.